### PR TITLE
Add the FIG secretary bylaw changes

### PR DESCRIPTION
**Pull request by michaelcullum**
Please comment in-line for any spelling or grammatical mistakes. Conceptual discussion should take please on the [Mailing List topic](https://groups.google.com/forum/#!topic/php-fig/oWFkNWBnjHQ).

Thanks

### CODE REVIEW

File: bylaws/001-voting-protocol.md [hunk 1]
```diff
@@ -38,6 +38,9 @@ to apply it immediately for reasons outside the purview of this group.
 
         - Regarding point 7, it is binding on all members.
 
+9. Responsibility for voting tallying is shared between the member who starts a vote
```
Comment (michaelcullum) [inline bylaws/001-voting-protocol.md]: Ah thanks

File: bylaws/001-voting-protocol.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 Voting Protocol
 ===============
 
-1. Only voting members may initiate a call for votes on a measure.
+1. Only voting members and FIG secretaries may initiate a call for votes on a measure.
```
Comment (michaelcullum) [inline bylaws/001-voting-protocol.md]: Ah thanks.

**Comment by michaelcullum** [comment]
@Chippiewill It shows where I've actually changed text as the highlight is a darker green. Sorry about the line endings though, that was just my editor.

**Comment by dragoonis** [comment]
@michaelcullum remember to ping me when everyone is happy with this, for a final pre-merge review :)

**Comment by michaelcullum** [comment]
@dragoonis Yeah. This will need a vote anyways

**Comment by dragoonis** [comment]
Okay, good.

File: bylaws/003-membership.md [hunk 3]
```diff
@@ -128,5 +128,119 @@ available.
...
+vote (just as any member project), but not cast any votes. However it should be
+noted that any vote started by one FIG secretary must be managed and tallied by a
+different secretary to ensure impartiality.
+
+### Selection:
+
+Secretaries are selected every year at the end of January after a vote according
+to standard [Voting Protocol bylaw][voting] except with the change that any
+outgoing secretaries may also participate in the vote. Secretaries are then in
+post until such a time as they resign, are removed by a vote of confidence or
```
Comment (svpernova09) [inline bylaws/003-membership.md]: Phrasing here needs to be update "vote of no confidence" as used in other places.

File: bylaws/003-membership.md [hunk 4]
```diff
@@ -128,5 +128,119 @@ available.
...
+* Tallying votes
+* Tracking member project activity, and marking projects as inactive
+* Ensuring bylaws are being followed
+* Clarifying any interpretation of bylaw text (If there is lack of consensus
+between FIG secretaries then it must be put to a full vote according to the
+standard [Voting Protocol bylaw][voting]).
+* Ensure that relevant marketing mediums (e.g. Twitter and Facebook) are kept
+professional, up to date and impartial.
+* Moderate discussions on github, the mailing list and IRC channels to ensure
+that an appropriate environment is maintained
```
Comment (svpernova09) [inline bylaws/003-membership.md]: Should add a task here:

* Should there be a FIG meeting at a conference or other ad hoc gathering any Secretary in attendance should take notes to report back to the mailing list.

**Comment by svpernova09** [comment]
Discussed using Instant-runoff voting for Secretary selection at the FIG meeting at phpworld

**Comment by michaelcullum** [comment]
@svpernova09 Instant-runoff only works when selecting a single winner. When electing more than one person in a single election STV (Basically instant-runoff but for multiple winners) works a lot better.

For anyone unaware of how STV works (in two rather differing formats depending on how you prefer to learn about this kind of stuff): https://www.youtube.com/watch?v=l8XOZJkozfI or https://en.wikipedia.org/wiki/Single_transferable_vote

File: bylaws/003-membership.md (hunk 4 above)
Comment (michaelcullum) [inline bylaws/003-membership.md]: Done but GH hasn't picked up on it.

File: bylaws/003-membership.md [hunk 5]
```diff
@@ -128,5 +128,121 @@ available.
...
+such as vote management should always be assigned to the FIG secretary.
+
+All FIG secretaries are expected to remain impartial and professional when acting
+in an official capacity as secretary and should also remain aware that even when
+not acting in an official capacity, their actions reflect back on the PHP FIG and
+the FIG must not be brought into disrepute.
+
+### Defined functions
+
+There are a number of defined functions that the FIG secretaries are expected to
```
Comment (bradynpoulsen) [inline bylaws/003-membership.md]: Should "secretaries" be uppercase or lowercase (as also mentioned above)?

File: bylaws/003-membership.md [hunk 6]
```diff
@@ -128,5 +128,121 @@ available.
...
+### Selection:
+
+Secretaries are selected every year at the end of January after a vote according
+to standard [Voting Protocol bylaw][voting] except with the change that any
+outgoing secretaries may also participate in the vote. Secretaries are then in
+post until such a time as they resign, are removed by a vote of no confidence or
+until the following January.
+
+At all times there should be at one or more FIG secretary from the previous term
+standing for re-election.
```
Comment (bradynpoulsen) [inline bylaws/003-membership.md]: Should this read "be at **least** one or more"?

Comment (michaelcullum) [inline bylaws/003-membership.md]: `at least one` or `one or more`  would be correct but not both. But thanks for pointing this out.

File: bylaws/003-membership.md (hunk 5 above)
Comment (michaelcullum) [inline bylaws/003-membership.md]: throughout we don't really capitalise it but there it was at the beginning of a sentence so made sense

File: bylaws/003-membership.md [hunk 7]
```diff
@@ -128,5 +128,129 @@ available.
...
+
+All FIG Secretaries are expected to remain impartial and professional when acting
+in an official capacity as secretary and should also remain aware that even when
+not acting in an official capacity, their actions reflect back on the PHP FIG and
+the FIG must not be brought into disrepute.
+
+### Defined functions
+
+There are a number of defined functions that the FIG Secretaries are expected to
+complete but they may also, within their remit defined in the above overaching
```
Comment (mvriel) [inline bylaws/003-membership.md]: s/overaching/overarching

File: bylaws/003-membership.md [hunk 8]
```diff
@@ -128,5 +128,129 @@ available.
...
+### Access and Abilities
+
+FIG Secretaries will be given access to the GitHub organisation as 'Owners' and
+full (admin) access to anything relating to official website, marketing and
+communication mediums including but not limited to the domain, twitter, IRC
+channel, packagist packages and mailing list in order to allow them to complete
+their duties and ensure that FIG mediums are managed by representatives elected
+by the PHP FIG.
+
+Also, whilst not full members themseleves, they have the ability to start any
```
Comment (mvriel) [inline bylaws/003-membership.md]: s/themseleves/themselves/

File: bylaws/003-membership.md (hunk 7 above)
Comment (michaelcullum) [inline bylaws/003-membership.md]: You never know. It may be an aching role as well.

Thanks.

Comment (mvriel) [inline bylaws/003-membership.md]: It probably will be. But no reason to make people wiser than they already are in advance, right? ;)

File: bylaws/003-membership.md [hunk 9]
```diff
@@ -128,5 +128,129 @@ available.
...
+
+* Keeping the website, twitter and other marketing mediums up-to-date
+* GitHub organization administration
+* Tallying votes
+* Tracking member project activity, and marking projects as inactive
+* Ensuring bylaws are being followed
+* Clarifying any interpretation of bylaw text (If there is lack of consensus
+between FIG Secretaries then it must be put to a full vote according to the
+standard [Voting Protocol bylaw][voting]).
+* Ensure that relevant marketing mediums (e.g. Twitter and Facebook) are kept
```
Comment (saltybeagle) [inline bylaws/003-membership.md]: This is a duplicate of the first function, https://github.com/php-fig/fig-standards/pull/673/files#diff-985a6593928ed8328147a01582d93604R160 suggest removing or combining the two, including impartiality.

File: bylaws/003-membership.md [hunk 10]
```diff
@@ -128,5 +128,129 @@ available.
...
+professional, up to date and impartial.
+* Moderate discussions on github, the mailing list and IRC channels to ensure
```
Comment (webmozart) [inline bylaws/003-membership.md]: GitHub (consistent with above)

File: bylaws/003-membership.md (hunk 9 above)
Comment (webmozart) [inline bylaws/003-membership.md]: :+1: (also "twitter" above vs. "Twitter" here - should be capital IMO)

File: bylaws/003-membership.md [hunk 11]
```diff
@@ -128,5 +128,129 @@ available.
...
+that an appropriate environment is maintained
+* Should there be a FIG meeting at a conference or other ad hoc gathering any
+Secretary in attendance should take notes to report back to the mailing list.
+
+### Access and Abilities
+
+FIG Secretaries will be given access to the GitHub organisation as 'Owners' and
+full (admin) access to anything relating to official website, marketing and
+communication mediums including but not limited to the domain, twitter, IRC
```
Comment (webmozart) [inline bylaws/003-membership.md]: "Twitter"

File: bylaws/003-membership.md [hunk 12]
```diff
@@ -128,5 +128,129 @@ available.
...
+channel, packagist packages and mailing list in order to allow them to complete
+their duties and ensure that FIG mediums are managed by representatives elected
+by the PHP FIG.
+
+Also, whilst not full members themselves, they have the ability to start any
+vote (just as any member project), but not cast any votes. However it should be
+noted that any vote started by one FIG Secretary must be managed and tallied by a
+different secretary to ensure impartiality.
+
+### Selection:
```
Comment (webmozart) [inline bylaws/003-membership.md]: the trailing colon is superfluous here and inconsistent with the other headings

File: bylaws/003-membership.md [hunk 13]
```diff
@@ -128,5 +128,129 @@ available.
...
+
+In the event of a resignation of a FIG Secretary or a vote of no confidence, they
+shall be removed with immediate effect and an existing FIG Secretary should
+commence a vote for a new secretary according to the aforementioned protocols.
+In the event of a resignation a special election will be performed following a
+similar procedure to regular elections under the administration of the remaining
+FIG Secretaries. The newly elected secretary would serve the remainder of the term
+that the outgoing secretary would have completed.
+
+### Eligibility Criteria:
```
Comment (webmozart) [inline bylaws/003-membership.md]: same here

**Comment by Crell** [comment]
This passed, didn't it? Shouldn't it be merged by now?

**Comment by michaelcullum** [comment]
Yep. Still waiting for it to be merged by someone (Both me and Korvin
request that be done so on the ML)

--
Michael C

**Comment by dragoonis** [comment]
@michaelcullum I've merged

**Comment by dragoonis** [comment]
merged.

**Pull request merged**